docs/guides/batching.md

```diff
@@ -0,0 +1,189 @@
+
+## Scenario
+
+Catered lunches and healthy snacks are some of the free perks startups offer to office employees, but with the rise of remote work, many startups are leaving this tradition.
+
+One way to keep the tradition going would be a universal meal delivery application in Slack, available as a `/ship-meal` command. Managers, anywhere in the world, could use the command to have lunch delivered to teammates, including those that work from home.
+
+### Multiple Vendors
+
+![image](../../static/images/meal-delivery-app.drawio.png)
+
+### Constraints
+
+The nature of such a service could cause traffic spikes for every upstream vendor needed to fulfil each individual order, so some care has to be taken when designing the meal app's API.
+For instance, if thousands of managers across the world use the command at the same time (just before lunch break) to place team orders, the sudden traffic spike will spread to upstream vendors, leading to delayed or failed orders.
+
+_Batching_ is a one technique that can be used to avoid overwhelming upstream servers with too many simultaneous requests. Batching combines multiple operations in a bulk operation that is sent in a single request.
+
+Tailcall supports batching via two [operators](https://tailcall.run/docs/operators/): `@upstream` and `@http`.
+Before we go over this Tailcall feature, we'll briefly review the most common implementations of batching in REST APIs.
+
+---
+
+## Batching in REST APIs
+
+### High Correspondence Between REST and CRUD
+
+In backends that adhere to the REST architectural style, the HTTP methods `POST`, `GET`, `PUT`, and `DELETE` roughly correspond to _Create_, _Read_, _Update_, and _Delete_ operations respectively in the CRUD paradigm as can be seen in the table below.
+
+| HTTP method | CRUD paradigm |
+| ----------- | ------------- |
+| `POST`      | _Create_      |
+| `GET`       | _Read_        |
+| `PUT`       | _Update_      |
+| `DELETE`    | _Delete_      |
+
+This one-to-one correspondence works because CRUD and REST often deal with a single entity or resource as the case may be.
+
+```bash
+  POST 	/v1/employees      (Create an employee entity)
+   GET 	/v1/employees/:id  (Read an employee entity)
+   PUT 	/v1/employees/:id  (Update an employee entity)
+DELETE 	/v1/employees/:id  (Delete an employee entity)
+   GET 	/v1/employees      (Read multiple employee entities)
+```
+
+### Low Correspondence Between Batching and CRUD
+
+The one-to-one correspondence doesn't carry over when batching is added to a REST API because batching can either involve:
+
+- performing the same operation on different entities of the same type (e.g. _Update_ the team's order so meals are shipped to employees with the following ids `1`, `4` & `7`) or;
+- grouping together different operations in one request (e.g. _Create_ `Jain` as a new employee, _Update_ an employee's meal preferences and _Delete_ meals above a certain price from the menu).
+
+### Real-world Examples of Batching
```

Comment:
Table isn't readable because we have too many columns. Flatten it into multiple paras.